File: sprint5/C++/lesson4code1.cpp
void insert_node(Node* root, int key) {
  if (key < root->key) {
    if (root->left == nullptr) {
      root->left = new Node(key);
    } else {
      insert_node(root->left, key);
    }
  } else {
    if (root->right == nullptr) {
      root->right = new Node(key);
    } else {
      insert_node(root->right, key);
    }
  }
}

File: sprint5/kotlin/lesson4code1.kt
fun insertNode(root: Node, key: Int) {
    if (key < root.key) {
        if (root.left == null) {
            root.left = Node(key)
        } else {
            insertNode(root.left, key)
        }
    } else {
        if (root.right == null) {
            root.right = Node(key)
        } else {
            insertNode(root.right, key)
        }
    }
}
